internals/overlord/servstate/handlers.go

```diff
@@ -337,16 +354,38 @@ func (s *serviceData) startInternal() error {
 	s.cmd = exec.Command(args[0], args[1:]...)
 	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	// Copy environment to avoid updating original.
-	environment := make(map[string]string)
-	for k, v := range s.config.Environment {
-		environment[k] = v
+	var environment map[string]string
+	if len(s.config.Environment) > 0 {
+		environment = maps.Clone(s.config.Environment)
+	} else {
+		environment = make(map[string]string)
+	}
```

Comment:
I think you can replace this block with:

```
environment := make(map[string]string)
maps.Copy(environment, s.config.Environment)
```

Copy seems to handle a nil source.